src/main/scala/io/github/pauljamescleary/petstore/config/DatabaseConfig.scala

```diff
@@ -1,32 +1,38 @@
 package io.github.pauljamescleary.petstore.config
 
-import cats.effect.{Async, Sync}
+import cats.effect.{Async, ContextShift, Resource, Sync}
 import doobie.hikari.HikariTransactor
 import org.flywaydb.core.Flyway
 
+import scala.concurrent.ExecutionContext
+
 case class DatabaseConfig(url: String, driver: String, user: String, password: String)
 
 object DatabaseConfig {
-
-  def dbTransactor[F[_]: Async](dbConfig: DatabaseConfig): F[HikariTransactor[F]] =
-    HikariTransactor.newHikariTransactor[F](dbConfig.driver, dbConfig.url, dbConfig.user, dbConfig.password)
+  def dbTransactor[F[_]: Async : ContextShift](
+    dbc: DatabaseConfig,
+    connEc : ExecutionContext,
+    transEc : ExecutionContext
+  ): Resource[F, HikariTransactor[F]] =
+    HikariTransactor.newHikariTransactor[F](dbc.driver, dbc.url, dbc.user, dbc.password, connEc, transEc)
 
   /**
     * Runs the flyway migrations against the target database
     *
     * This only gets applied if the database is H2, our local in-memory database.  Otherwise
     * we skip this step
     */
-  def initializeDb[F[_]](dbConfig: DatabaseConfig, xa: HikariTransactor[F])(
-      implicit S: Sync[F]): F[Unit] =
+  def initializeDb[F[_]](dbConfig: DatabaseConfig)(implicit S: Sync[F]): F[Unit] =
     if (dbConfig.url.contains(":h2:")) {
-      xa.configure { ds =>
-        S.delay {
-          val fw = new Flyway()
-          fw.setDataSource(ds)
-          fw.migrate()
-          ()
-        }
+      S.delay {
```

Comment:
So, I think we landed that we are going to remove the `if`, pass in a `datasource`, and always run, correct?  Or perhaps just pass in the Transactor?

Reply:
yep, i was going to split the changes needed for #102 out of this one, but it was a little annoying. I'll merge and update the other branch now.

Reply:
sorry, i shouldn't have let the two be dependent on each other that way.